src/openseadragon.js

```diff
@@ -685,19 +685,10 @@
   * @param {OpenSeadragon.Options} options - Viewer options.
   * @returns {OpenSeadragon.Viewer}
   */
-window.OpenSeadragon = window.OpenSeadragon || function( options ){
-
+function OpenSeadragon( options ){
```

Comment:
Ideally one shouldn't pollute global namespace. For example in my use-case original motivation for this change was necessity to use multiple version of openseadragon at the same time (in adition to openseadragon 2.* we're also using mirador which is still using openseadragon 1.*).

And before this change there was a conflict because both version of the library tried to use global. So ideally one should use something similar to snippet that you mentioned.

For example in other libraries one need to explicitly register extension with library instance. For example we're using https://github.com/cytoscape/cytoscape.js which provides reach set of extensions, there one need to explicitly register extension e.g. https://github.com/cytoscape/cytoscape.js-navigator/blob/master/cytoscape-navigator.js#L843